src/Utils.ts

```diff
@@ -422,25 +424,17 @@ export const Utils = {
       minDifferenceBetweenMatches: config.nicknameToLoginMinimumDifferenceBetweenMatches
     }): tm.Player | undefined {
     const players = PlayerService.players
-    const strippedNicknames: { strippedNickname: string, player: tm.Player }[] = []
+    const strippedNicknames = []
     for (const e of players) {
-      strippedNicknames.push({ strippedNickname: this.stripSpecialChars(Utils.strip(e.nickname).toLowerCase()), player: e })
-    }
-    const matches: { player: tm.Player, value: number }[] = []
-    for (const e of strippedNicknames) {
-      const value = dsc.twoStrings(e.strippedNickname, nickname.toLowerCase())
-      if (value > options.similarityGoal) {
-        matches.push({ player: e.player, value })
-      }
+      strippedNicknames.push(this.stripSpecialChars(Utils.strip(e.nickname)))
     }
+    const matches = []
+    const indices = this.matchString(nickname, strippedNicknames)
+    matches.push(indices.map(a => players[a]))
     if (matches.length === 0) {
       return undefined
     }
-    const s = matches.sort((a, b): number => b.value - a.value)
-    if (s[0].value - s?.[1]?.value ?? 0 < options.minDifferenceBetweenMatches) {
-      return undefined
-    }
-    return s[0].player
+    return matches[0][0]
```

Comment:
I'm not sure this is the best solution. Maybe check if there are two people with the same nick and then ask the mod to clarify which one to kick?

Reply:
i dont see this ever becoming an issue unless there are 2 people with the same exact or very similar names. tbh, its their fault not ours in that case.

as for the confirmation message, that seems like something chatservice should do, not a random util...